src/core/network/qgsblockingnetworkrequest.cpp

```diff
@@ -247,7 +247,7 @@ QgsBlockingNetworkRequest::ErrorCode QgsBlockingNetworkRequest::doRequest( QgsBl
         waitConditionMutex.unlock();
         break;
       }
-      authRequestBufferNotEmpty.wait( &waitConditionMutex );
+      //authRequestBufferNotEmpty.wait( &waitConditionMutex );
```

Comment:
maybe you could try the 2 arguments wait() with the mutex and a timeout (I believe in millisecond) : https://doc.qt.io/qt-5/qwaitcondition.html#wait-1
And add a comment in the code why that timeout is needed

Reply:
I don't understand the real reasons, but it seems like the idea with `sleep` or modified `wait` is failing #7055 :(
maybe like this:
`
QgsApplication::instance()->processEvents( QEventLoop::WaitForMoreEvents );